Doc2Vec version 4.0 breaks our Doc2VecVectorizer

Currently we have no restrictions for gensim versions so when gensim
recently released v4.0 which contains breaking changes it broke our
Doc2VecVectorizer. The error is

```
>       super(Doc2Vec, self).__init__(
            sentences=corpus_iterable,
            corpus_file=corpus_file,
            vector_size=self.vector_size,
            sg=(1 + dm) % 2,
            null_word=self.dm_concat,
            callbacks=callbacks,
            window=window,
            epochs=epochs,
            **kwargs,
        )
E       TypeError: __init__() got an unexpected keyword argument 'window_size'
```

So this could be as simple as changing the `window_size` to `window` or
more compicated. It will be good to pin the version to either < 4.0 or
>= 4.0 <5.0 when we fix this.

--- a/setup.py
+++ b/setup.py
@@ -50,7 +50,7 @@ setuptools.setup(
         'umap-learn',
         'nervaluate',
         'twine',
-        'gensim',
+        'gensim<5.0.0',
         # 'hdbscan',
         'cython',
         'flake8',

--- a/wellcomeml/ml/doc2vec_vectorizer.py
+++ b/wellcomeml/ml/doc2vec_vectorizer.py
@@ -87,7 +87,7 @@ class Doc2VecVectorizer(BaseEstimator, TransformerMixin):
         self.model = Doc2Vec(
             tagged_documents,
             vector_size=self.vector_size,
-            window_size=self.window_size,
+            window=self.window_size,
             workers=workers,
             min_count=self.min_count,
             epochs=self.epochs,

